orchagent/pfc_detect_mellanox.lua

```diff
@@ -63,27 +76,6 @@ for i = n, 1, -1 do
                 local pfc_rx_pkt_key = 'SAI_PORT_STAT_PFC_' .. queue_index .. '_RX_PKTS'
                 local pfc_duration_key = 'SAI_PORT_STAT_PFC_' .. queue_index .. '_RX_PAUSE_DURATION_US'
 
-                -- Get port specific timestamp
-                local port_timestamp_current = tonumber(redis.call('HGET', counters_table_name .. ':' .. port_id, 'PFC_WD_time_stamp'))
-                if port_timestamp_current ~= nil then
-                    local port_timestamp_lasttime = port_timestamp_last_cache[port_id]
-                    if port_timestamp_lasttime == nil then
-                        port_timestamp_lasttime = tonumber(redis.call('HGET', counters_table_name .. ':' .. port_id, 'PFC_WD_time_stamp_last'))
-                        port_timestamp_last_cache[port_id] = port_timestamp_lasttime 
-                        redis.call('HSET', counters_table_name .. ':' .. port_id, 'PFC_WD_time_stamp_last', port_timestamp_current)
-                    end
-
-                    if port_timestamp_lasttime ~= nil then
-                        effective_poll_time = (port_timestamp_current - port_timestamp_lasttime) / 1000
-                    else
-                        effective_poll_time = global_effective_poll_time
-                    end
-                    effective_poll_time_lasttime = false
-                else
-                    effective_poll_time = global_effective_poll_time
-                    effective_poll_time_lasttime = global_effective_poll_time_lasttime
-                end
-
                 -- Get all counters
                 local occupancy_bytes = redis.call('HGET', counters_table_name .. ':' .. KEYS[i], 'SAI_QUEUE_STAT_CURR_OCCUPANCY_BYTES')
```

Comment:
@stephenxs should we move this to HGETALL and get all counters for the queue and use that cache? this will also reduce queries inside the loop.

Reply:
Thanks @kperumalbfn 
I tend to keep it as it is.
This indeed reduces a hash lookup. However, we can hardly say whether it can yield any performance improvements without testing it.
In addition we need extra logic to convert the resulting list into a dictionary in Lua. Such logic introduces overheads and makes it difficult to understand.

Previously we measured the performance of PFC watchdog on a 256 port SKU. It is very fast to execute the whole Lua plugin part for all queues, including the communication time.
 
We can optimize it if we see it takes more time in the future but for now we don't need to do that.
Thanks

Reply:
Thanks. Can we update this as an improvement? For 512 port SKUs, this can have some impact.

Reply:
Thanks.
We also verified 512-port SKUs and it's the same.
But we will do it if it becomes a bottleneck.